### PR TITLE
Attach location to constants

### PR DESCRIPTION
**Pull request by gpetiot**
No diff with test_branch.sh

This allows us to remove all the code that parses the tokens extracted from Source.ml to retrieve the location.
Upstream PR: https://github.com/ocaml/ocaml/pull/10544

### CODE REVIEW

File: vendor/ocaml-4.13/ast_mapper.ml [hunk 1]
```diff
@@ -486,7 +490,8 @@ module P = struct
     | Ppat_var s -> var ~loc ~attrs (map_loc sub s)
     | Ppat_alias (p, s) -> alias ~loc ~attrs (sub.pat sub p) (map_loc sub s)
     | Ppat_constant c -> constant ~loc ~attrs (sub.constant sub c)
-    | Ppat_interval (c1, c2) -> interval ~loc ~attrs c1 c2
+    | Ppat_interval (c1, c2) ->
+        interval ~loc ~attrs (sub.constant sub c1) (sub.constant sub c2)
```
Comment (gpetiot) [inline vendor/ocaml-4.13/ast_mapper.ml:494]: This should be upstreamed, I will open a PR

**Comment by gpetiot** [comment]
I think we can assume this change won't be upstreamed so it's ready to be reviewed here.

File: vendor/ocaml-4.13-extended/parser.mly [hunk 2]
```diff
@@ -55,6 +55,7 @@ let mkmod ~loc ?attrs d = Mod.mk ~loc:(make_loc loc) ?attrs d
 let mkstr ~loc d = Str.mk ~loc:(make_loc loc) d
 let mkclass ~loc ?attrs d = Cl.mk ~loc:(make_loc loc) ?attrs d
 let mkcty ~loc ?attrs d = Cty.mk ~loc:(make_loc loc) ?attrs d
+let mkcst ~loc c = Const.mk ~loc:(make_loc loc) c
```
Comment (jberdine) [inline vendor/ocaml-4.13-extended/parser.mly:58]: Nit, but aren't the other functions named `mk<foo>` where the corresponding constructors are `P<foo>`, indicating that `mkconst` would be a more symmetric name than `mkcst`?

**Review by jberdine — APPROVED** [review]
This is a nice simplification! The code looks good to me.

File: vendor/ocaml-4.13-extended/parser.mly (hunk 2 above)
Comment (gpetiot) [reply vendor/ocaml-4.13-extended/parser.mly]: Done!

**Comment by gpetiot** [comment]
I will wait for #1783 to be merged before merging this one

**Review by Julow — APPROVED** [review]
Looks good :)

**Pull request merged**